db/migrations/0022.add_lti_type.py

```diff
@@ -0,0 +1,28 @@
+'''
+Add in the initial LTI Type Table
+'''
+
+from yoyo import step
+
+__depends__ = {'0020.add_lti_usage'}
+
+steps = [
+   step('''
+        CREATE TABLE IF NOT EXISTS lti_type (
+            canvas_id INTEGER NOT NULL,
+            des VARCHAR(200),
+            rpt_group_nm VARCHAR(200),
+            PRIMARY KEY (canvas_id)
+        ) ENGINE = InnoDB DEFAULT CHARSET = utf8mb4;
+    '''),
+    step('''
+        ALTER TABLE lti_placement
+        ADD FOREIGN KEY (placement_type_id) REFERENCES lti_type(canvas_id) 
+        ON DELETE RESTRICT ON UPDATE CASCADE
```

Comment:
According to "MySQL 8.0 Reference Manual  /  ...  /  FOREIGN KEY Constraints" (https://dev.mysql.com/doc/refman/8.0/en/create-table-foreign-keys.html#foreign-key-referential-actions):

> RESTRICT: Rejects the delete or update operation for the parent table. Specifying RESTRICT (or NO ACTION) is the same as omitting the ON DELETE or ON UPDATE clause.

The 5.7 manual says the same.  I think our DBs are 5.7.

Reply:
'Explicit is better than implicit' right? ;) 

I think I'd still need the clause there for ON UPDATE as I think the default for that is NO ACTION, and I do want it to update if those are updated.

Reply:
Yeah, explicitness usually doesn't hurt.  Unless MySQL's functionality changes later.

I think `on update` and `on delete` can be specified independently.  At least, that's what I gather from the syntax shown in the manual.

Anyway, no big deal.  Resolving…

Reply:
Thanks, it's nice to know the defaults. I didn't think to look those up and these were the values I felt most appropriate. Though realistically neither of these probably matter either way.